bats/tests/containers/split-dns-vpn.bats

```diff
@@ -0,0 +1,50 @@
+# bats file_tags=opensuse
+
+load '../helpers/load'
+
+REGISTRY_URL=$(echo "$RD_VPN_TEST_IMAGE" | cut -d'/' -f1)
+
+local_setup() {
+    if ! using_vpn_url; then
+        skip "This test requires a connection to the designated VPN."
+    fi
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+@test 'Can access private registry over VPN from host' {
+    run curl -I -k "https://$REGISTRY_URL/v2"
+    assert_success
+    # We avoid assert_line here due to carriage return (\r) issues in curl output.
+    # The output format differs slightly between Ubuntu's and openSUSE's curl packages,
+    # so we use a partial match instead for consistency.
+    assert_output --partial "docker-distribution-api-version: registry/2.0"
+}
+
+@test 'Can pull image from private registry over VPN' {
+    run ctrctl pull --quiet "$RD_VPN_TEST_IMAGE"
+    assert_success
+}
+
+@test 'Can verify container access to the registry' {
+    run ctrctl run --rm "$RD_VPN_TEST_IMAGE" curl -I -k "https://$REGISTRY_URL/v2"
+    assert_success
+    assert_output --partial "docker-distribution-api-version: registry/2.0"
+}
+
+verify_host_connectivity() {
+    run ctrctl run --rm "$RD_VPN_TEST_IMAGE" sh -c "zypper install --no-confirm iputils && timeout -s INT 10 ping -c 5 $1"
```

Comment:
I don't think we should expect the test image to be based on openSUSE, so calling `zypper` feels wrong.

We don't need to use the image pulled from the VPN to run the ping test:

```suggestion
    run ctrctl run --rm "$IMAGE_BUSYBOX" timeout -s INT 10 ping -c 5 host.rancher-desktop.internal
```